### PR TITLE
Add generate-drawing-examples tool

### PR DESCRIPTION
**Pull request by rfuest**
This PR adds a new `generate-drawing-examples` crate to the tools directory. It is an replacement for the previous tool in the `simulator` crate, which currently causes some dependency problems. Because of this problems the tool currently uses e-g `0.7.0-alpha.2` instead of a later release.

The new tool should be easier to use, because examples can now be written as normal functions and the `{}` hack for newlines isn't necessary anymore. Comments inside the example code are now also preserved, which wasn't the case in the old tool, but I hadn't noticed this problem until now.

### CODE REVIEW

**Comment by jamwaffles** [comment]
Will the dependency issues be resolved once embedded-graphics/simulator#21 is merged?

File: justfile [hunk 1]
```diff
@@ -73,6 +76,26 @@ check-links: generate-docs
...
+# Generate drawing examples in the doc directory
+generate-drawing-examples:
+    cd tools/generate-drawing-examples && cargo run
+
+# Checks if drawing examples are up to date
+check-drawing-examples: generate-drawing-examples
+    git diff --quiet doc/ || ( \
+        echo "doc/ folder is not up to date" \
+        echo "Try running 'just generate-drawing-examles'." \
+        echo "If any images have changed, run just generate-drawing-examples-montage' to update the collage image too" \
```
Comment (jamwaffles) [inline justfile:88]:
```suggestion
        echo "If any images have changed, run 'just generate-drawing-examples-montage' to update the collage image too" \
```

**Review by jamwaffles — CHANGES_REQUESTED** [review]
A tiny nit on an error message, otherwise LGTM. Very clever use of macros!

File: justfile (hunk 1 above)
Comment (rfuest) [reply justfile]: Good catch! This suggestion also made me notice a typo in the line above.

**Comment by rfuest** [comment]
> Will the dependency issues be resolved once [embedded-graphics/simulator#21](https://github.com/embedded-graphics/simulator/pull/21) is merged?

Yes, I think they should be resolved after that PR is merged. We should release a new simulator alpha after that PR.

**Review by jamwaffles — APPROVED** [review]
:+1:

**Pull request merged**